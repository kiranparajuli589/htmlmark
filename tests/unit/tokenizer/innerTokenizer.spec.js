const outerTokenizer = require("../../../lib/tokenizer/outerTokenizer")
const innerTokenizer = require("../../../lib/tokenizer/innerTokenizer")

describe("inner tokenizer", () => {
    describe("no deep tokenization (newline, codeblock, comment, image)", () => {
        it.each([
            ["some code", ""],
            ["some code", "\n"]
          ])("should not deep tokenize new lines", (lines) => {
            const tokenizedContent = outerTokenizer(lines)
            const deepTokenizedContent = innerTokenizer(tokenizedContent)
            expect(deepTokenizedContent[1]).toEqual(tokenizedContent[1])
          })
          it.each([
            [
                // codeblock
                "```js",
                "const a = 1",
        
                "const b = 2",
                "```",
            ],
            ["<!-- some comment -->"], // comment
            ["![alt text](urlpart)"] // image
          ])("should not tokenize", (lines) => {
            const tokenizedContent = outerTokenizer(lines)
            const result = innerTokenizer(tokenizedContent)
            expect(result).toEqual(tokenizedContent)
          })
    })
    describe("quote", () => {
        it.each([
            {quote: "> quote 1", expectedDepth: 0},
            {quote: "> > quote 2", expectedDepth: 1},
            {quote: "> > > quote 3", expectedDepth: 2},
            {quote: "> > > > quote 4", expectedDepth: 3},
            {quote: "> > > > > quote 5", expectedDepth: 4},
        ])("should detect the quote depth", ({quote, expectedDepth}) => {
            const tokenizedContent = outerTokenizer([quote])
            const deepTokenizedContent = innerTokenizer(tokenizedContent)
            expect(deepTokenizedContent[0].depth).toEqual(expectedDepth)
        })
        it.each([
            "> quote **one** with *two*",
            "  > quote `four` with ~~five~~",
            "> quote [link-title](link-url) with *two*",
        ])("should deep tokenize quote with zero depth", (line) => {
            const tokenizedContent = outerTokenizer([line])
            const deepTokenizedContent = innerTokenizer(tokenizedContent)
            expect(deepTokenizedContent).toMatchSnapshot()
        })
        it.each([
            "> > quote **one** with *two*",
            "  > > > quote `four` with ~~five~~",
            "> > > > quote [link-title](link-url) with *two*",
        ])("should deep tokenize quote with multiple depth", (line) => {
            const tokenizedContent = outerTokenizer([line])
            const deepTokenizedContent = innerTokenizer(tokenizedContent)
            expect(deepTokenizedContent).toMatchSnapshot()
        })
    })

    describe("paragraph", () => {
        it("should be deep tokenized", () => {
          const lines = [
            "a paragraph of words `first code` normal text here `code body` *first italics* here me crying *italic body* here me crying **first bolds** some normal again **bold body** [Kiran Parajuli](https://kiranparajuli.com.np) ~~strikes body~~ here some"
          ]
          const tokenizedContent = outerTokenizer(lines)
          const deepTokenizedContent = innerTokenizer(tokenizedContent)
          expect(deepTokenizedContent).toMatchSnapshot()
        })
    })
})