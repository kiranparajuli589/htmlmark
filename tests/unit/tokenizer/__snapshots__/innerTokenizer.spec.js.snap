// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`inner tokenizer paragraph should be deep tokenized 1`] = `
Array [
  Object {
    "raw": "a paragraph of words \`first code\` normal text here \`code body\` *first italics* here me crying *italic body* here me crying **first bolds** some normal again **bold body** [Kiran Parajuli](https://kiranparajuli.com.np) ~~strikes body~~ here some",
    "tokens": Array [
      Object {
        "type": "text",
        "value": "a paragraph of words ",
      },
      Object {
        "type": "code",
        "value": "first code",
      },
      Object {
        "type": "text",
        "value": " normal text here ",
      },
      Object {
        "type": "code",
        "value": "code body",
      },
      Object {
        "type": "text",
        "value": " ",
      },
      Object {
        "type": "italic",
        "value": "first italics",
      },
      Object {
        "type": "text",
        "value": " here me crying ",
      },
      Object {
        "type": "italic",
        "value": "italic body",
      },
      Object {
        "type": "text",
        "value": " here me crying ",
      },
      Object {
        "type": "bold",
        "value": "first bolds",
      },
      Object {
        "type": "text",
        "value": " some normal again ",
      },
      Object {
        "type": "bold",
        "value": "bold body",
      },
      Object {
        "type": "text",
        "value": " ",
      },
      Object {
        "href": "https://kiranparajuli.com.np",
        "type": "link",
        "value": "Kiran Parajuli",
      },
      Object {
        "type": "text",
        "value": " ",
      },
      Object {
        "type": "strike",
        "value": "strikes body",
      },
      Object {
        "type": "text",
        "value": " here some",
      },
    ],
    "type": "paragraph",
  },
]
`;

exports[`inner tokenizer quote should deep tokenize quote with multiple depth 1`] = `
Array [
  Object {
    "depth": 1,
    "indent": "",
    "raw": "> > quote **one** with *two*",
    "tokens": Array [
      Object {
        "type": "text",
        "value": "quote ",
      },
      Object {
        "type": "bold",
        "value": "one",
      },
      Object {
        "type": "text",
        "value": " with ",
      },
      Object {
        "type": "italic",
        "value": "two",
      },
    ],
    "type": "quote",
  },
]
`;

exports[`inner tokenizer quote should deep tokenize quote with multiple depth 2`] = `
Array [
  Object {
    "depth": 2,
    "indent": "  ",
    "raw": "  > > > quote \`four\` with ~~five~~",
    "tokens": Array [
      Object {
        "type": "text",
        "value": "quote ",
      },
      Object {
        "type": "code",
        "value": "four",
      },
      Object {
        "type": "text",
        "value": " with ",
      },
      Object {
        "type": "strike",
        "value": "five",
      },
    ],
    "type": "quote",
  },
]
`;

exports[`inner tokenizer quote should deep tokenize quote with multiple depth 3`] = `
Array [
  Object {
    "depth": 3,
    "indent": "",
    "raw": "> > > > quote [link-title](link-url) with *two*",
    "tokens": Array [
      Object {
        "type": "text",
        "value": "quote ",
      },
      Object {
        "href": "link-url",
        "type": "link",
        "value": "link-title",
      },
      Object {
        "type": "text",
        "value": " with ",
      },
      Object {
        "type": "italic",
        "value": "two",
      },
    ],
    "type": "quote",
  },
]
`;

exports[`inner tokenizer quote should deep tokenize quote with zero depth 1`] = `
Array [
  Object {
    "depth": 0,
    "indent": "",
    "raw": "> quote **one** with *two*",
    "tokens": Array [
      Object {
        "type": "text",
        "value": "quote ",
      },
      Object {
        "type": "bold",
        "value": "one",
      },
      Object {
        "type": "text",
        "value": " with ",
      },
      Object {
        "type": "italic",
        "value": "two",
      },
    ],
    "type": "quote",
  },
]
`;

exports[`inner tokenizer quote should deep tokenize quote with zero depth 2`] = `
Array [
  Object {
    "depth": 0,
    "indent": "  ",
    "raw": "  > quote \`four\` with ~~five~~",
    "tokens": Array [
      Object {
        "type": "text",
        "value": "quote ",
      },
      Object {
        "type": "code",
        "value": "four",
      },
      Object {
        "type": "text",
        "value": " with ",
      },
      Object {
        "type": "strike",
        "value": "five",
      },
    ],
    "type": "quote",
  },
]
`;

exports[`inner tokenizer quote should deep tokenize quote with zero depth 3`] = `
Array [
  Object {
    "depth": 0,
    "indent": "",
    "raw": "> quote [link-title](link-url) with *two*",
    "tokens": Array [
      Object {
        "type": "text",
        "value": "quote ",
      },
      Object {
        "href": "link-url",
        "type": "link",
        "value": "link-title",
      },
      Object {
        "type": "text",
        "value": " with ",
      },
      Object {
        "type": "italic",
        "value": "two",
      },
    ],
    "type": "quote",
  },
]
`;
